soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -67,50 +82,93 @@ static ALWAYS_INLINE void clock_init(void)
 #if defined(CONFIG_SOC_LPC55S06) || defined(CONFIG_SOC_LPC55S16) || \
 	defined(CONFIG_SOC_LPC55S28) || defined(CONFIG_SOC_LPC55S36) || \
 	defined(CONFIG_SOC_LPC55S69_CPU0)
-    /*!< Set up the clock sources */
-    /*!< Configure FRO192M */
-	/*!< Ensure FRO is on  */
+	/* Set up the clock sources */
+	/* Configure FRO192M */
+	/* Ensure FRO is on  */
 	POWER_DisablePD(kPDRUNCFG_PD_FRO192M);
-	/*!< Set up FRO to the 12 MHz, just for sure */
+	/* Set up FRO to the 12 MHz, to ensure we can change the clock freq */
 	CLOCK_SetupFROClocking(12000000U);
-	/*!< Switch to FRO 12MHz first to ensure we can change the clock */
+	/* Switch to FRO 12MHz first to ensure we can change the clock */
 	CLOCK_AttachClk(kFRO12M_to_MAIN_CLK);
 
+	/* Configure XTAL32M */
+#if !defined(CONFIG_SOC_LPC55S36)
+	POWER_DisablePD(kPDRUNCFG_PD_XTAL32M);
+	POWER_DisablePD(kPDRUNCFG_PD_LDOXO32M);
+#endif
+	CLOCK_SetupExtClocking(16000000U);
+
+	/*!< Ensure CLK_IN is on  */
+	SYSCON->CLOCK_CTRL |= SYSCON_CLOCK_CTRL_CLKIN_ENA_MASK;
+	ANACTRL->XO32M_CTRL |= ANACTRL_XO32M_CTRL_ENABLE_SYSTEM_CLK_OUT_MASK;
+
+
+#if defined(CONFIG_SOC_LPC55S06) || !defined(CONFIG_INIT_PLL1)
 	/* Enable FRO HF(96MHz) output */
 	CLOCK_SetupFROClocking(96000000U);
 
+	/* Switch MAIN_CLK to FRO_HF */
+	CLOCK_AttachClk(kFRO_HF_to_MAIN_CLK);
+
+	SystemCoreClock = 96000000U;
+#else
+	/* Switch PLL1 clock source selector to XTAL32M */
+	CLOCK_AttachClk(kEXT_CLK_to_PLL1);
+
+	/* Ensure PLL1 is on */
+	POWER_DisablePD(kPDRUNCFG_PD_PLL1);
+	POWER_DisablePD(kPDRUNCFG_PD_PLL1_SSCG);
+
+	/* Configure PLL to the desired values */
+	CLOCK_SetPLL1Freq(&pll1Setup);
+
+	CLOCK_SetClkDiv(kCLOCK_DivPll1Clk, 0U, true);
+	CLOCK_SetClkDiv(kCLOCK_DivPll1Clk, 1U, false);
+	CLOCK_AttachClk(kPLL1_to_MAIN_CLK);
+
+	SystemCoreClock = 150000000U;
+#endif /* CONFIG_SOC_LPC55S06 || !CONFIG_INIT_PLL1 */
+
+
+#if !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+#if defined(CONFIG_SOC_LPC55S06)
+	/* Set Voltage for one of the fastest clock outputs: System clock output */
+	POWER_SetVoltageForFreq(96000000U);
+	/*!< Set FLASH wait states for core */
+	CLOCK_SetFLASHAccessCyclesForFreq(96000000U);
+#else
+	/* Set Voltage for one of the fastest clock outputs: System clock output */
+	POWER_SetVoltageForFreq(150000000U);
+	/*!< Set FLASH wait states for core */
+	CLOCK_SetFLASHAccessCyclesForFreq(150000000U);
+#endif /* CONFIG_SOC_LPC55S06 */
+#endif /* !CONFIG_TRUSTED_EXECUTION_NONSECURE */
+
+
 #ifdef CONFIG_INIT_PLL0
-	/*!< Ensure XTAL16M is on  */
+	/* Ensure XTAL32M is on  */
 	PMC->PDRUNCFGCLR0 |= PMC_PDRUNCFG0_PDEN_XTAL32M_MASK;
 	PMC->PDRUNCFGCLR0 |= PMC_PDRUNCFG0_PDEN_LDOXO32M_MASK;
```

Comment:
This fails to build on `lpcxpresso55s36` because these bit masks are different.  Also, this seems redundant with the earlier call to CLOCK_SetupExtClocking()